browser/browsing_data/brave_browsing_data_remover_delegate.cc

```diff
@@ -91,3 +110,68 @@ void BraveBrowsingDataRemoverDelegate::ClearShieldsSettings(
     }
   }
 }
+
+#if BUILDFLAG(IPFS_ENABLED)
+void BraveBrowsingDataRemoverDelegate::WaitForIPFSRepoGC(
+    base::Process process) {
+  bool exited = false;
+
+  {
+    base::ScopedAllowBaseSyncPrimitives scoped_allow_base_sync_primitives;
+
+    // Because we set maximum IPFS storage size as 1GB in Brave, ipfs repo gc
+    // command should be finished in just a few seconds and we do not expect
+    // this child process would hang forever. To be safe, we will wait for 30
+    // seconds max here.
+    exited = process.WaitForExitWithTimeout(base::TimeDelta::FromSeconds(30),
+                                            nullptr);
+  }
+
+  if (!exited)
+    process.Terminate(0, false /* wait */);
+}
+
+// Run ipfs repo gc command to clear IPFS cache when IPFS executable path is
+// available. Because the command does not support time ranged cleanup, we will
+// always clear the whole cache expect for pinned files when clearing browsing
+// data.
+void BraveBrowsingDataRemoverDelegate::ClearIPFSCache() {
+  auto* service =
+      ipfs::IpfsServiceFactory::GetInstance()->GetForContext(profile_);
+  if (!service)
+    return;
+
+  base::FilePath path = service->GetIpfsExecutablePath();
+  if (path.empty())
```

Comment:
We should get the last known good path when the executable path is not determined here please in a follow up.

Reply:
Thanks, opened https://github.com/brave/brave-browser/issues/13605.